src/Core/Console/Application.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Console;
+
+use AppKernel;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Bundle\FrameworkBundle\Console\Application as FrameworkConsoleApplication;
+
+/**
+ * The PrestaShop Console application
+ */
+final class Application extends FrameworkConsoleApplication
+{
+    public function __construct(KernelInterface $kernel)
+    {
+        parent::__construct($kernel);
+
+        $this->setName('PrestaShop');
+        $this->setVersion(AppKernel::VERSION);
+
+        $inputDefinition = $this->getDefinition();
+        $inputDefinition->addOption(new InputOption('employee', '-em', InputOption::VALUE_REQUIRED, 'Specify employee context (id).', null));
+
+        // @see MultiShopCommandListener BC compatible fix to display the options in the Console
```

Comment:
It's nice you're trying to solve all issues at once 😄 but I think it's a big challenge
Maybe we should aim to fix the different contexts in the console with different PRs ? I suggest handling the multishop in a separate PR, I'm sure it's a little bomb 💣 that will require a lot of efforts to tackle

Reply:
it doesn't support multishop feature per see, it only displays the available options for each command :)

it's up to every command to handle or not these options ('id_shop' and 'id_shop_group') and it's probably a little :bomb: I do agree